File: Barath/src/12_CapitalizeFirstLetter/capitalizeFirstLetter.test.ts
// capitalizeFirstLetter.test.ts

import { capitalizeFirstLetter } from './capitalizeFirstLetter';

describe('capitalizeFirstLetter', () => {

    // Test with a sentence containing multiple words
    test('should capitalize the first letter of each word in a sentence', () => {
        const input = "the quick brown fox";
        const result = capitalizeFirstLetter(input);
        expect(result).toBe("The Quick Brown Fox");
    });

    // Test with a single word
    test('should capitalize the first letter of a single word', () => {
        const input = "hello";
        const result = capitalizeFirstLetter(input);
        expect(result).toBe("Hello");
    });

    // Test with an empty string
    test('should return an empty string when input is empty', () => {
        const input = "";
        const result = capitalizeFirstLetter(input);
        expect(result).toBe("");
    });
});

File: Bipin/012_capitalize_first_letter/capitalize_first_letter_test.go
package main

import (
	"testing"
)

// Test function for capitalizeFirstLetter
func TestCapitalizeFirstLetter(t *testing.T) {
	// Test case 1: Sentence with multiple words
	result := CapitalizeFirstLetter("the quick brown fox")
	expected := "The Quick Brown Fox"
	if result != expected {
		t.Errorf("Test Case 1 Failed: expected '%s', got '%s'", expected, result)
	}

	// Test case 2: Single word
	result = CapitalizeFirstLetter("hello")
	expected = "Hello"
	if result != expected {
		t.Errorf("Test Case 2 Failed: expected '%s', got '%s'", expected, result)
	}

	// Test case 3: String with numbers and special characters
	result = CapitalizeFirstLetter("123 hello! world")
	expected = "123 Hello! World"
	if result != expected {
		t.Errorf("Test Case 3 Failed: expected '%s', got '%s'", expected, result)
	}

	// Test case 4: Empty string
	result = CapitalizeFirstLetter("")
	expected = ""
	if result != expected {
		t.Errorf("Test Case 4 Failed: expected '%s', got '%s'", expected, result)
	}
}
